doc/guides/misc/how_to_doc.md

```diff
@@ -0,0 +1,156 @@
+---
+title: Creating Guides & How Starlight Works
+description: Explanation on how the starlight "build system works and how to create guides
+code_folder: examples/basic/hello-world/
+---
+
+## What is Starlight?
+
+[Starlight](https://starlight.astro.build/) is a documentation site generator built on top of [Astro](https://astro.build/), a popular modern web framework, by the Astro team itself.
+Starlight is designed to be fast, flexible, and easy to use, making it a great choice for creating documentation sites. It supports Markdown and MDX, allowing you to write content in a familiar format while also enabling extensive access to the entire modern web ecosystem if needed.
+
+For RIOT Starlight is meant to exclusively host guides while the API documentation is generated by Doxygen and hosted on the [RIOT API documentation](https://doc.riot-os.org/) site.
+Where Doxygen has a very automated approach to generating documentation, Starlight is meant to be more flexible and allow for a more human touch that is needed for tutorials and guides,
+which means that we both have the control over the content and layout but at the same time have to put more active effort into the UX of the documentation itself.
+
+## How to run the site locally
+
+To run the site locally, you need to have [Node.js](https://nodejs.org/) installed.
+It is highly recommended to use a version manager such as [NVM](https://github.com/nvm-sh/nvm) to manage your Node.js versions.
+
```

Comment:
What is the minimum version? Either point to the config file that specifies it (preferred, cause the guide can't go out of sync that easily) or write it down directly.